File: Startupba/Startupba.Services/Database/Payment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Startupba.Services.Database
{
    /// <summary>
    /// A Stripe payment record. Linked to a Donation once the payment is confirmed.
    /// </summary>
    public class Payment
    {
        [Key]
        public int Id { get; set; }

        public int? DonationId { get; set; }

        [Required]
        [MaxLength(255)]
        public string StripePaymentIntentId { get; set; } = string.Empty;

        [MaxLength(255)]
        public string? StripeCustomerId { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        [Required]
        [MaxLength(10)]
        public string Currency { get; set; } = "EUR";

        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "pending";

        [MaxLength(50)]
        public string? PaymentMethod { get; set; }

        [MaxLength(255)]
        public string? CustomerName { get; set; }

        [MaxLength(255)]
        public string? CustomerEmail { get; set; }

        [MaxLength(500)]
        public string? BillingAddress { get; set; }

        [MaxLength(100)]
        public string? BillingCity { get; set; }

        [MaxLength(100)]
        public string? BillingState { get; set; }

        [MaxLength(100)]
        public string? BillingCountry { get; set; }

        [MaxLength(20)]
        public string? BillingZipCode { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        // Navigation property
        [ForeignKey("DonationId")]
        public Donation? Donation { get; set; }
    }
}

File: Startupba/UI/startupba_desktop/lib/model/payment.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment.g.dart';

@JsonSerializable()
class Payment {
  final int id;
  final int? donationId;
  final String stripePaymentIntentId;
  final String? stripeCustomerId;
  final double amount;
  final String currency;
  final String status;
  final String? paymentMethod;
  final String? customerName;
  final String? customerEmail;
  final String? billingAddress;
  final String? billingCity;
  final String? billingState;
  final String? billingCountry;
  final String? billingZipCode;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final String startupName;
  final int? userId;
  final String userName;

  Payment({
    this.id = 0,
    this.donationId,
    this.stripePaymentIntentId = '',
    this.stripeCustomerId,
    this.amount = 0,
    this.currency = 'eur',
    this.status = '',
    this.paymentMethod,
    this.customerName,
    this.customerEmail,
    this.billingAddress,
    this.billingCity,
    this.billingState,
    this.billingCountry,
    this.billingZipCode,
    required this.createdAt,
    this.updatedAt,
    this.startupName = '',
    this.userId,
    this.userName = '',
  });

  factory Payment.fromJson(Map<String, dynamic> json) =>
      _$PaymentFromJson(json);
  Map<String, dynamic> toJson() => _$PaymentToJson(this);
}
